Sources/SwiftKuery/Connection.swift

```diff
@@ -124,59 +197,109 @@ public protocol Connection {
+    /// - Returns: A QueryResult representing the result of the operation
+    func releaseSync(savepoint: String) -> QueryResult
 }
 
 public extension Connection {
+
     func execute(query: Query, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        do {
-            let databaseQuery = try query.build(queryBuilder: queryBuilder)
-            let (convertedQuery, namedToNumbered, count) = Utils.convertNamedParametersToNumbered(query: databaseQuery, queryBuilder: queryBuilder)
-            var numberedParameters: [Any?] = Array(repeating: nil, count: count)
-            for (parameterName, parameterValue) in parameters {
-                if let numbers = namedToNumbered[parameterName] {
-                    for number in numbers {
-                        numberedParameters[number - 1] = parameterValue
+        DispatchQueue.global().async {
+            do {
+                let databaseQuery = try query.build(queryBuilder: self.queryBuilder)
+                let (convertedQuery, namedToNumbered, count) = Utils.convertNamedParametersToNumbered(query: databaseQuery, queryBuilder: self.queryBuilder)
+                var numberedParameters: [Any?] = Array(repeating: nil, count: count)
+                for (parameterName, parameterValue) in parameters {
+                    if let numbers = namedToNumbered[parameterName] {
+                        for number in numbers {
+                            numberedParameters[number - 1] = parameterValue
+                        }
+                    }
+                    else {
+                        onCompletion(.error(QueryError.syntaxError("Failed to map parameters.")))
                     }
                 }
-                else {
-                    onCompletion(.error(QueryError.syntaxError("Failed to map parameters.")))
-                }
+                self.execute(convertedQuery, parameters: numberedParameters, onCompletion: onCompletion)
+            }
+            catch  QueryError.syntaxError(let error) {
+                onCompletion(.error(QueryError.syntaxError(error)))
+            }
+            catch {
+                onCompletion(.error(QueryError.syntaxError("Failed to build the query.")))
             }
-            execute(convertedQuery, parameters: numberedParameters, onCompletion: onCompletion)
         }
-        catch  QueryError.syntaxError(let error) {
-            onCompletion(.error(QueryError.syntaxError(error)))
+    }
+
+    func executeSync(query: Query, parameters: [String:Any?]) -> QueryResult {
+        var result: QueryResult?
+        let semaphore = DispatchSemaphore(value: 0)
+        execute(query: query, parameters: parameters) { res in
+            result = res
+            semaphore.signal()
         }
-        catch {
-            onCompletion(.error(QueryError.syntaxError("Failed to build the query.")))
+        semaphore.wait()
+        guard let resultUnwrapped = result else {
```

Comment:
Style is just `guard let result = result else {`

Reply:
This doesn't work as there is already a local called result defined.